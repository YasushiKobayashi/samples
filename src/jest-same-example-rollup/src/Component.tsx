import * as React from 'react'
import type { render } from '@testing-library/react'

export const Component: React.FC = () => {
  return <div>test</div>
}

type TestUtils = {
  render: typeof render
}

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const testUtils: TestUtils = {}

/* @__PURE__ */
describe('Component', () => {
  beforeEach(async () => {
    const { render } = await import('@testing-library/react')
    testUtils.render = render
  })

  afterEach(async () => {
    const { cleanup } = await import('@testing-library/react')
    cleanup()
  })

  it('render test', async () => {
    const { container, asFragment } = testUtils.render(<Component />)
    expect(container.textContent).toContain('test')
    expect(asFragment()).toMatchSnapshot()
  })
})
