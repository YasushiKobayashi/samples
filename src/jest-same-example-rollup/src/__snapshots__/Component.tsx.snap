// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render test 1`] = `
<DocumentFragment>
  <div>
    test
  </div>
</DocumentFragment>
`;
